### PR TITLE
PAYARA-4077: Implement Weld SecurityContext

### PR DESCRIPTION
**Pull request by pdudits**
This enable security context propagation to asynchronous event observers.

This is build on top of PAYARA-4078 to make CDI TCK pass.

### CODE REVIEW

**Comment by pdudits** [comment]
Jenkins test please

**Comment by pdudits** [comment]
jenkins test please

**Comment by pdudits** [comment]
Jenkins test please

**Comment by pdudits** [comment]
Jenkins test please

**Comment by pdudits** [comment]
jenkins test please

**Comment by pdudits** [comment]
jenkins test please

**Comment by pdudits** [comment]
Jenkins test please

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java [hunk 1]
```diff
@@ -110,13 +110,7 @@
...
 import org.glassfish.web.deployment.descriptor.ServletFilterMappingDescriptor;
 import org.glassfish.weld.connector.WeldUtils;
-import org.glassfish.weld.services.EjbServicesImpl;
-import org.glassfish.weld.services.ExternalConfigurationImpl;
-import org.glassfish.weld.services.InjectionServicesImpl;
-import org.glassfish.weld.services.NonModuleInjectionServices;
-import org.glassfish.weld.services.ProxyServicesImpl;
-import org.glassfish.weld.services.SecurityServicesImpl;
-import org.glassfish.weld.services.TransactionServicesImpl;
+import org.glassfish.weld.services.*;
```
Comment (dmatej) [inline appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java]: Whe should agree on some rule how many imports should be replaced with asterisk ... I have 20 as default ... and I would not like it under 10. Only minor comment here, does not block PR.

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/ResourceInjectionServicesImpl.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    private static final Logger LOG = Logger.getLogger(ResourceInjectionServicesImpl.class.getName());
+
+
+    private final InitialContext namingContext;
+
+    public ResourceInjectionServicesImpl() {
+        try {
+            this.namingContext = new InitialContext();
+        } catch (NamingException e) {
+            throw new RuntimeException("Initial context not available at deployment time", e);
```
Comment (dmatej) [inline appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/ResourceInjectionServicesImpl.java]: Use IllegalStateException, don't use RTE directly ...

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java (hunk 1 above)
Comment (Pandrex247) [reply appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java]: Yeah IntelliJ defaults to something like 5 which I think is way too low.
We don't currently have one set, but I agree we should agree one.

Personally I hate wildcard imports in all situations.
You pretty much **never** need to import everything, and IDEs can auto-hide imports anyway so the "readability" argument is moot.

Comment (dmatej) [reply appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java]: Yep, SonarQube also has "too many imports" rule ... so usually asterisk should be unreachable :)

Comment (pdudits) [reply appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java]: It was set to 9 :). I'll turn it up a bit.

**Pull request merged**